lib/internal/net.js

```diff
@@ -9,3 +9,9 @@ function isLegalPort(port) {
     return false;
   return +port === (port >>> 0) && port >= 0 && port <= 0xFFFF;
 }
+
+
+function assertPort(port) {
+  if (typeof port !== 'undefined' && !isLegalPort(port))
```

Comment:
Looking at the code where `assertPort` is being used, we see:

```
        assertPort(h.port);
        if (h.host)
          listenAfterLookup(h.port | 0, h.host, backlog, h.exclusive);
        else
          listen(self, null, h.port | 0, 4, backlog, undefined, h.exclusive);
```

Notice that `listen` is called with `h.port | 0`, so that when `h.port` is undefined, `assertPort` falls through and it defaults to `0`... specifically to account for when h.port is not explicitly specified.